jproperties-checks/src/main/java/org/sonar/jproperties/checks/generic/HardCodedCredentialsCheck.java

```diff
@@ -38,18 +42,56 @@
 @ActivatedByDefault
 public class HardCodedCredentialsCheck extends DoubleDispatchVisitorCheck {
 
+  private static final String DEFAULT_ENCRYPTED_CREDENTIALS_TO_IGNORE = "";
+
   private static final Pattern HARD_CODED_USERNAME = Pattern.compile(".*(login|username).*", Pattern.CASE_INSENSITIVE);
   private static final Pattern HARD_CODED_PASSWORD = Pattern.compile(".*(password|passwd|pwd).*", Pattern.CASE_INSENSITIVE);
 
+  @RuleProperty(
+    key = "encryptedCredentialsToIgnore",
+    description = "Regular expression of encrypted credentials to ignore. "
+      + "See " + CheckUtils.LINK_TO_JAVA_REGEX_PATTERN_DOC + " for detailed regular expression syntax. "
+      + "For example, to ignore encrypted credentials starting with 'ENC(' and 'OBF:', set the parameter to '^(ENC\\(|OBF:).+$'. "
+      + "Leave empty if encrypted credentials should not be ignored.",
+    defaultValue = DEFAULT_ENCRYPTED_CREDENTIALS_TO_IGNORE)
+  private String encryptedCredentialsToIgnore = DEFAULT_ENCRYPTED_CREDENTIALS_TO_IGNORE;
+
   @Override
-  public void visitKey(KeyTree tree) {
-    if (HARD_CODED_USERNAME.matcher(tree.text()).matches()) {
-      addPreciseIssue(tree, "Remove this hard-coded username.");
+  public void visitProperty(PropertyTree tree) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) The Cyclomatic Complexity of this method "visitProperty" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)